pytest_examples/traceback.py

```diff
@@ -21,16 +20,17 @@ def create_example_traceback(exc: Exception, module_path: str, example: CodeExam
         # f_code.co_posonlyargcount was added in 3.8
         return None
     frames = []
-    for frame, _ in traceback.walk_tb(exc.__traceback__):
+    tb = exc.__traceback__
+    while tb is not None:
+        frame = tb.tb_frame
         if frame.f_code.co_filename == module_path:
-            frames.append(create_custom_frame(frame, example))
+            frames.append((create_custom_frame(frame, example), tb.tb_lasti, tb.tb_lineno + example.start_line))
```

Comment:
I think we should fix `create_custom_frame` rather than work around it.

Reply:
I don't know if it was broken, it seemed that frame.f_lineno was correct, although I don't understand how. The point is that we don't want to use f_lineno at all, from neither the original nor the altered frame.

I've updated `create_custom_frame` to only modify the filename in the code object, just to make it clear.